### PR TITLE
support chrome-headless & firefox-headless -- option 1

### PR DESCRIPTION
**Pull request by mach6**
- [X] By placing an `X` in the preceding checkbox, I verify that I am a PayPal employee or
I have signed the [Contributor License Agreement](https://github.com/paypal/SeLion#contributing)

---

- support chrome-headless & firefox-headless
- deprecate phantomjs, selendroid, and ios-driver
- update travis build to use latest firefox for repo clones and sauce labs for parent repo
- update travis build to use chrome-headless and firefox-headless
- update travis build -- chrome tests are no longer allowed to fail

### CODE REVIEW

**Comment by mach6** [comment]
@sebady @sundaramrajendran this one should be good for review.

File: client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/HeadlessChromeCapabilitiesBuilder.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        // existing.get("args") is an immutable list.
+        // so we need to declare a new one and copy the values in, when present
+        List<String> args = Lists.newArrayList();
+        if (existing.containsKey("args")) {
+            args.addAll(existing.get("args"));
+        }
+
+        args.remove("--headless");
+        args.add("--headless");
```
Comment (sundaramrajendran) [inline client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/HeadlessChromeCapabilitiesBuilder.java:41]: @mach6 : Above line args got removed and adding same args again. I think it is by mistake?

**Review by sundaramrajendran — COMMENTED** [review]
@mach6 : Overall looks good, except one review comments.

File: client/src/main/java/com/paypal/selion/internal/platform/grid/BrowserFlavors.java [hunk 2]
```diff
@@ -23,12 +23,15 @@
  */
 public enum BrowserFlavors {
     FIREFOX("*firefox"),
+    FIREFOX_HEADLESS("*firefox-headless"),
```
Comment (sebady) [inline client/src/main/java/com/paypal/selion/internal/platform/grid/BrowserFlavors.java:26]: Have you considered making "HeadLess" primarily a global config option ?  To me it fits neatly as a global option right along side the "runLocally" config option etc and less so as a Browser "flavor".

**Comment by mach6** [comment]
Another possible approach is submitted as PR #376

File: client/src/main/java/com/paypal/selion/internal/platform/grid/BrowserFlavors.java (hunk 2 above)
Comment (mach6) [reply client/src/main/java/com/paypal/selion/internal/platform/grid/BrowserFlavors.java:26]: See PR #376

**Comment by mach6** [comment]
closing this. we went with a different approach

**Pull request closed without merging**